code/modules/mob/living/carbon/damage_procs.dm

```diff
@@ -248,26 +248,65 @@
 	var/update = 0
-	while(parts.len && (brute > 0 || burn > 0 || stamina > 0))
-		var/obj/item/bodypart/picked = pick(parts)
-		var/brute_per_part = round(brute/parts.len, DAMAGE_PRECISION)
-		var/burn_per_part = round(burn/parts.len, DAMAGE_PRECISION)
-		var/stamina_per_part = round(stamina/parts.len, DAMAGE_PRECISION)
+
+	///Okay, so, "What the fuck is going on here", you might be asking...
+	if(stamina)
+		//Build our list of bodyparts which are capable of holding stamina damage still.
+		for(var/obj/item/bodypart/BP as anything in bodyparts)
+			if(BP.stamina_dam < BP.max_stamina_damage)
+				parts += BP
+
+		///We need two lists to ensure a pseudo-even distribution of damage.
+		///This is our list of all parts that can still hold some more stamina damage.
+		var/list/obj/item/bodypart/not_full = parts.Copy()
+		while(stamina)
+			///Copy the total parts list. This is necessary because later we remove instances from this list during the loop.
+			parts = not_full.Copy()
+			///If all of our bodyparts are full on stamina, stop the loop.
+			if(!length(parts))
+				break
+
+			while(length(parts))
+				//Create a "share" of the current stamina damage pool
+				var/damage_share = round(stamina/length(parts), DAMAGE_PRECISION)
+				//Pick and take a part from the eligible parts list.
+				var/obj/item/bodypart/picked = pick_n_take(parts)
+				//If the part is full on stamina damage, remove it from the master list so it doesn't get re-added
+				if(picked.stamina_dam >= picked.max_stamina_damage)
+					not_full -= picked
+					continue
+				///Distribute the damage share to this limb, record how much was actually used.
+				var/stamina_was = picked.stamina_dam
+				update |= picked.receive_damage(0, 0, damage_share, FALSE, required_status, wound_bonus = CANT_WOUND)
+				stamina = round(stamina - (picked.stamina_dam - stamina_was), DAMAGE_PRECISION)
+				parts -= picked
+
+	var/list/obj/item/bodypart/not_full = get_damageable_bodyparts(required_status)
+	parts = not_full.Copy()
+	while((brute > 0 || burn > 0) && length(not_full))
+		if(!length(parts))
+			parts = not_full.Copy()
+
+		var/brute_per_part = round(brute/length(parts), DAMAGE_PRECISION)
+		var/burn_per_part = round(burn/length(parts), DAMAGE_PRECISION)
+
+		var/obj/item/bodypart/picked = pick_n_take(parts)
+		if(picked.get_damage() >= picked.max_damage)
+			not_full -= picked
+			continue
 
 		var/brute_was = picked.brute_dam
 		var/burn_was = picked.burn_dam
-		var/stamina_was = picked.stamina_dam
 
-
-		update |= picked.receive_damage(brute_per_part, burn_per_part, stamina_per_part, FALSE, required_status, wound_bonus = CANT_WOUND) // disabling wounds from these for now cuz your entire body snapping cause your heart stopped would suck
+		update |= picked.receive_damage(brute_per_part, burn_per_part, 0, FALSE, required_status, wound_bonus = CANT_WOUND) // disabling wounds from these for now cuz your entire body snapping cause your heart stopped would suck
```

Comment:
it's unclear what some of these parameters are actually corresponding to, can you name them like how you did with wound_bonus? I don't think you need to explicitly set parameters that have a default value of the values you are already putting in, like setting stamina damage to 0